### ISSUE TITLE
Hard timeout for sharded entity passivation

### ISSUE THREAD

**Issue by johanandren** [akka/akka#29964]
When passivation is done through a custom message, it will first have to wait for any pre-existing messages in the sharded actor inbox to be processed and then any specific logic implemented to do passivation, potentially not immediately stopping the entity. 

If the inbox queue is very long or the processing of each message takes a lot of time consequently passivation will take a lot of time, delaying shard shutdown and rebalance. Worse, if there is a bug in the passivation logic of the entity leading to it never stopping that will hinder ever rebalancing the shard.

Would it make sense to add a hard passivation timeout after which any passivation not completing triggers a kill of the entity?

**Comment by Zhen-hao** [comment]
it would indeed be nice to give users a parameter to control the timeout value, with the infinite option (can be value `0` or `-1`)

**Comment by johanandren** [comment]
Yeah, if we implement I think the default should be off, so an opt-in by defining a timeout, since it depends so much on the use case.
